src/main/java/frc/team2412/robot/util/Constants.java

```diff
@@ -0,0 +1,94 @@
+        public static final int[] frontRightTurningEncoderPorts = new int[] { 4, 5 };
+        public static final int[] rearRightTurningEncoderPorts = new int[] { 5, 6 };
+
+        public static final boolean frontLeftTurningEncoderReversed = false;
+        public static final boolean rearLeftTurningEncoderReversed = true;
+        public static final boolean frontRightTurningEncoderReversed = false;
+        public static final boolean rearRightTurningEncoderReversed = true;
+
+        public static final int[] frontLeftDriveEncoderPorts = new int[] { 7, 8 };
+        public static final int[] rearLeftDriveEncoderPorts = new int[] { 9, 10 };
+        public static final int[] frontRightDriveEncoderPorts = new int[] { 11, 12 };
+        public static final int[] rearRightDriveEncoderPorts = new int[] { 13, 14 };
+
+        public static final boolean frontLeftDriveEncoderReversed = false;
+        public static final boolean rearLeftDriveEncoderReversed = true;
+        public static final boolean frontRightDriveEncoderReversed = false;
+        public static final boolean rearRightDriveEncoderReversed = true;
+
+        public static final double trackWidth = 1.0;
+        // Distance between centers of right and left wheels on robot
+        public static final double wheelBase = 1.0;
+        // Distance between front and back wheels on robot
+        public static final SwerveDriveKinematics driveKinematics = new SwerveDriveKinematics(
+                new Translation2d(wheelBase / 2, trackWidth / 2),
+                new Translation2d(wheelBase / 2, -trackWidth / 2),
+                new Translation2d(-wheelBase / 2, trackWidth / 2),
+                new Translation2d(-wheelBase / 2, -trackWidth / 2));
+
+    }
+
+    public static final class ModuleConstants {
+
+    }
+
+    // todod someone must find these values
+    public static final class AutoConstants {
+        public static final double MaxSpeedMetersPerSecond = 3;
+        public static final double MaxAccelerationMetersPerSecondSquared = 3;
+        public static final double MaxAngularSpeedRadiansPerSecond = Math.PI;
+        public static final double MaxAngularSpeedRadiansPerSecondSquared = Math.PI;
+
+        public static final double PXController = 0.1;
+        public static final double PYController = 0.1;
+        public static final double PThetaController = 1;
+
+        // Constraint for the motion profilied robot angle controller
+        public static final TrapezoidProfile.Constraints kThetaControllerConstraints = new TrapezoidProfile.Constraints(
+                MaxAngularSpeedRadiansPerSecond, MaxAngularSpeedRadiansPerSecondSquared);
+
+        public static final double maxSpeedMetersPerSecond = 0.3;
+        public static final double maxAccelerationMetersPerSecondSquared = 0.3;
+        public static final double maxAngularSpeedRadiansPerSecond = Math.PI;
+        public static final double maxAngularSpeedRadiansPerSecondSquared = Math.PI;
+
+    }
```

Comment:
Since these are all only ever used for the AutonomousCommand, I'd recommend moving them into a static class in that file and then deleting Constants.java

Reply:
And I'd recommend updating the names to match the naming convention used elsewhere (UPPERCASE_CONSTANT)

Reply:
sounds good